openwisp_controller/config/base/config.py

```diff
@@ -291,12 +291,14 @@ def _send_config_modified_signal(self):
         Emits ``config_modified`` signal.
         Called also by Template when templates of a device are modified
         """
-        config_modified.send(
-            sender=self.__class__,
-            instance=self,
-            # kept for backward compatibility
-            config=self,
-            device=self.device,
+        transaction.on_commit(
+            config_modified.send(
```

Comment:
`transaction.on_commit` gets passed a callable (a function), you are calling the function instead, which means it will be executed immediately and `None` will be passed to `transaction.on_commit`.

You have to pass a `lambda` expression (which is an anonymous funcion in python) in which `config_modified.send` is called, as in the example I provided: https://github.com/openwisp/openwisp-firmware-upgrader/blob/4200399d3d71a0995896819f6259ebae9db9a64d/openwisp_firmware_upgrader/base/models.py#L86-L88